src/lib/dnssd/Advertiser.h

```diff
@@ -204,6 +215,15 @@ class CommissionAdvertisingParameters : public BaseAdvertisingParams<CommissionA
     }
     CommissioningMode GetCommissioningMode() const { return mCommissioningMode; }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+    CommissionAdvertisingParameters & SetJointFabricMode(std::optional<JointFabricMode> mode)
```

Comment:
Should take `optional<BitFlags<JointFabricMode>>`, I would think, given that multiple flags can be set.

Because for example 14 (an example given in the spec) is not a valid value of JointFabricMode the way it's defined here.

At that point presumably you don't even need the `optional`, in that you can just use empty BitFlags to represent "none of the bits are set, don't list the key".